internal/controller/factory/statefulset.go

```diff
@@ -32,8 +33,62 @@ import (
 	etcdaenixiov1alpha1 "github.com/aenix-io/etcd-operator/api/v1alpha1"
 )
 
-const (
-	etcdContainerName = "etcd"
+type certConfigurator interface {
+	Name() string
+	MountPath() string
+	CrtFilePath() string
+	KeyFilePath() string
+}
+
+type certificate struct {
+	certName  string
+	mountPath string
+	crtFile   string
+	keyFile   string
+}
+
+type certificates map[string]certConfigurator
+
+type etcdConfigurator interface {
+	ContainerName() string
+	VolumeName() string
+	MountPath() string
+	DirName() string
+	DataPath() string
+}
+
+type etcd struct {
+	containerName string
+	volumeName    string
+	mountPath     string
+	dirName       string
+}
```

Comment:
IMO, it is premature to create a completely private struct and an interface wrapper around it. At the moment I don't understand the need for an abstraction like this, but an extra layer of indirection will make the code less readable.

Also I'm having trouble understanding the purpose of these structs. In our context `etcd` is quite a generic name. The fields/methods of these data structures hint at volumes, rather than an etcd cluster in general. A documentation comment would be quite helpful, as would a more specific name.